Source/VMware.vSphereDSC/VMware.vSphereDSC.build.ps1

```diff
@@ -76,26 +97,12 @@ if (Test-Path -Path $script:DSCResourcesFolder) {
     $resources = "'{0}'" -f $resources
     $dscResourcesToExport = "DscResourcesToExport = @($resources)"
 
-    $index = 1
-    $startLine = 0
-    $endLine = 0
-    $counter = 0
-
     $psdFileContent = Get-Content -Path $script:PsdPath
-    $psdFileContent | ForEach-Object {
-        if ($_ -Like '*DscResourcesToExport*') {
-            $startLine = $index
-            $counter++
-        }
-
-        if ($_.Trim().EndsWith(')') -and $counter -ne 0) {
-            $endLine = $index
-            $counter = 0
-        }
 
-        $index++
-    }
+    $range = Get-LinesRange -FileContent $psdFileContent -StartLinePattern '*DscResourcesToExport*' -EndLinePattern ')'
+    $startLine = $range.Get_Item("startLine")
+    $endLine = $range.Get_Item("endLine")
 
     $psdFileContent = $psdFileContent[0..($startLine - 2)], $dscResourcesToExport, $psdFileContent[$endLine..($psdFileContent.Length - 1)]
```

Comment:
this seems to be wrong it should be

[0..($startLine -1)], $dscResourceLine, [$endLine + 1.. Length - 1]

for example
0 a
1 b
2 c
3  DSCResouces = @(a,b,c,d)
4 e
5 f

0..(3-1)
a
b
c
here is the replaces line
(3+1)..(6-1) is
e
f